src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -184,12 +184,16 @@ public String createSnapshot(
       // We fetch the DUOS dataset to confirm its existence, but do not need the returned value.
       duosClient.getDataset(duosId, userReq);
     }
+
+    UUID snapshotId = UUID.randomUUID();
```

Comment:
This is fine but you could inline it on line 196 too, I don't think a variable is needed here.

Reply:
Yes I can add this change!